src/Meziantou.Analyzer/Rules/ReturnTaskDirectlyAnalyzer.cs

```diff
@@ -0,0 +1,233 @@
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Meziantou.Analyzer.Internals;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Meziantou.Analyzer.Rules;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class ReturnTaskDirectlyAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly DiagnosticDescriptor s_rule = new(
+        RuleIdentifiers.ReturnTaskDirectly,
+        title: "Return Task instead of awaiting it",
+        messageFormat: "Return Task instead of awaiting it",
+        RuleCategories.Performance,
+        DiagnosticSeverity.Info,
+        isEnabledByDefault: true,
+        description: "Returning a Task directly instead of awaiting it increases performance, as the compiler does not need to create an additional state machine.",
+        helpLinkUri: RuleIdentifiers.GetHelpUri(RuleIdentifiers.ReturnTaskDirectly));
+
+    public override void Initialize(AnalysisContext context)
+	{
+		context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+		context.EnableConcurrentExecution();
+		context.RegisterCompilationStartAction(OnCompilationStart);
+	}
+
+	private static void OnCompilationStart(CompilationStartAnalysisContext context)
+    {
+        var analyzerContext = new AnalyzerContext(context.Compilation);
+        context.RegisterSyntaxNodeAction(ctx => OnMethodAnalysis(ctx, analyzerContext), SyntaxKind.MethodDeclaration, SyntaxKind.LocalFunctionStatement, SyntaxKind.ParenthesizedLambdaExpression);
```

Comment:
You can move `OnMethodAnalysis` into the `AnalyzerContext` class and remove the `static` modifier. It will simplify the code as you can use the properties directly.

```suggestion
        context.RegisterSyntaxNodeAction(ctx => analyzerContext.OnMethodAnalysis(ctx), SyntaxKind.MethodDeclaration, SyntaxKind.LocalFunctionStatement, SyntaxKind.ParenthesizedLambdaExpression);
```